docs/commands/dev.md

```diff
@@ -30,6 +30,7 @@ netlify dev
 - `live` (*boolean*) - start a public live session
 - `offline` (*boolean*) - disables any features that require network access
 - `port` (*string*) - port of netlify dev
+- `prod` (*boolean*) - build the site for production and serve locally
```

Comment:
Just creeping on your PR 😄  How does this work with or in contrast to the context flag (documented above), that you can also set to `production` ? 

ie `netlify dev --context production`

Reply:
(the CLI doc above mentions only environment variables, but our [docs site ](https://docs.netlify.com/cli/get-started/#get-started-with-netlify-dev) makes it sound like using the context flag builds your site for production - so this would need to be updated if that's not correct)

Reply:
Your creeping is very much welcome and that's a great point! The CLI docs are correct: the `context` flag only affects environment variables. This makes me wonder whether we should repurpose `context` and introduce this new behaviour (i.e. actually building the site for production) when `context` is `production`. And we could make `--prod` an alias for `--context=production`.

I think that would make things more consistency, with just two concerns worth considering:

1. That would make this a breaking change, since we'd be changing an existing flag rather than introducing a new one
2. This new behaviour means that watching for files won't work, and people will have to restart the CLI for the build to be run again and their changes to be picked up. I worry that it might be confusing that a specific value of `context` triggers this behaviour, rather than an entirely new flag.

But if we decide not to reuse `context`, it'll be a bit difficult to explain why you have `--prod` and `--context=production`.

@ascorbic What are your thoughts on this one?

Reply:
I thought about this a bit more and I feel that a more manageable step would be to keep `context` as is (i.e. just about environment variables) and make `prod` automatically set `--context=production`, so that you're building the production site and using the production env vars.

Reply:
Now that we have named the flag `prod`, I agree that making it automatically set `--context=production` feels most intuitive

Reply:
This is why I sugegsted having it as a separate command. Using `ntl dev` when it's building for production is a little unintuitive

Reply:
Building a separate command is a bigger lift (if done properly) that unfortunately I don't have the capacity for right now. I've pushed https://github.com/netlify/cli/pull/5338/commits/ae74853c17f5113dae181fe57230fef02fd3c7ea which sets `--context=production` when `--prod` is set, and in https://github.com/netlify/cli/pull/5338/commits/f3cc600d13d8aea1ac8efc75a7093ca99d2b502e added a "beta" label to the flag in the help menu. We do this for other commands.

I suggest that we ship this as is and then iterate on it. If we come to the conclusion that this makes more sense as a separate command, we can break it out and deprecate `--prod`.

Reply:
Yeah, that is a good compromise. Does our CLI lib support aliases? If so, that would probably be the simplest solution.

Reply:
https://github.com/netlify/cli/pull/5351 turns the flag into a separate command. We can get it merged after this one.